File: CyclicRotation.cs
using System;

class Solution {
    public int[] solution(int[] A, int K) {
        if (A.Length <=1)   
            return A;  
        
        for(int rotationStep = 0; rotationStep < K; rotationStep++)
        {
              int lastElement = A[A.Length-1];
               
               for(int arrayIndex = A.Length-1; arrayIndex > 0; arrayIndex--)
                   A[arrayIndex] = A[arrayIndex-1];
                   
                A[0] = lastElement;
        }
        return A;
    }
}

File: cyclic_rotation.py
def solution(A, K):
    if len(A) <= 1:
        return A
    for rotationStep in range(1, K+1):
        lastElement = A[len(A)-1]
        for index in range(1, len(A)+1):
            A[len(A)-index] = A[len(A)-index-1]
        A[0] = lastElement
    return A